### PR TITLE
Add `import_source_credential` RPC - use CodeBuild in `ecr_build` RPC

### PR DESCRIPTION
**Pull request by mtp1376**

### CODE REVIEW

File: src/modules/aws_codebuild/rpcs/import_source_credential.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  } as const;
+
+  /** @internal */
+  importSourceCredentials = crudBuilderFormat<CodeBuild, 'importSourceCredentials', string | undefined>(
+    'importSourceCredentials',
+    input => input,
+    res => res?.arn,
+  );
+
+  validServerTypes = ['BITBUCKET', 'GITHUB', 'GITHUB_ENTERPRISE'];
```
Comment (yrobla) [inline src/modules/aws_codebuild/rpcs/import_source_credential.ts:26]: this could be an enum

File: src/modules/aws_codebuild/rpcs/import_source_credential.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  validAuthTypes = ['BASIC_AUTH', 'OAUTH', 'PERSONAL_ACCESS_TOKEN'];
+
+  private makeError(message: string) {
+    return [
+      {
+        arn: '',
+        status: 'ERROR',
```
Comment (yrobla) [inline src/modules/aws_codebuild/rpcs/import_source_credential.ts:33]: we should unify at some point. I return OK/KO as there were other rpcs doing that. At some point we need to have a common standard for the returns of those functions

File: src/modules/aws_codebuild/rpcs/import_source_credential.ts [hunk 3]
```diff
@@ -0,0 +1,82 @@
+import { Build, CodeBuild, ImportSourceCredentialsInput } from '@aws-sdk/client-codebuild';
+import { ServerType } from '@aws-sdk/client-codebuild/dist-types/models/models_0';
+
+import { AWS, crudBuilderFormat } from '../../../services/aws_macros';
+import { Context, RpcBase, RpcResponseObject } from '../../interfaces';
+import { AwsCodebuildModule } from '../index';
+
+export class ImportSourceCredentialRpc extends RpcBase {
```
Comment (yrobla) [inline src/modules/aws_codebuild/rpcs/import_source_credential.ts:8]: can you add typedoc documentation as i did for other rpcs? so we keep the documentation up to date

File: src/modules/aws_codebuild/rpcs/import_source_credential.ts (hunk 2 above)
Comment (mtp1376) [reply src/modules/aws_codebuild/rpcs/import_source_credential.ts:33]: Agreed. Let's raise it in the channel to discuss furtuer.

File: examples/ecs-fargate/django/app/infra/migrations/0001_install.py [hunk 4]
```diff
@@ -14,12 +14,12 @@ class Migration(migrations.Migration):
         migrations.RunSQL(
             """
             SELECT * FROM iasql_install(
-                'aws_ecs_simplified'
+                'aws_ecs_simplified', 'aws_codebuild'
```
Comment (aguillenv) [inline examples/ecs-fargate/django/app/infra/migrations/0001_install.py:17]: why is this added if it is not in use?

File: src/modules/aws_ecr/rpcs/build.ts [hunk 5]
```diff
@@ -164,31 +89,44 @@ export class EcrBuildRpc extends RpcBase {
...
-    awsIamModule.role.cloud.delete(role, ctx);
+    await awsIamModule.role.cloud.delete(role, ctx);
 
     // delete credentials
-    if (githubPersonalAccessToken) await this.deleteGithubCredentials(credentialsArn, client);
+    if (githubPersonalAccessToken)
+      await awsCodebuildModule.sourceCredentialsList.cloud.delete(
+        { arn: credentialsArn } as SourceCredentialsList, // hacky but works
+        ctx,
+      );
```
Comment (aguillenv) [inline src/modules/aws_ecr/rpcs/build.ts:126]: can we add the braces here to avoid confusion?

File: src/modules/aws_ecr/rpcs/build.ts [hunk 6]
```diff
@@ -140,18 +56,27 @@ export class EcrBuildRpc extends RpcBase {
     githubRef: string,
     githubPersonalAccessToken: string,
   ): Promise<RpcResponseObject<typeof this.outputTable>[]> => {
-    await this.ensureAwsIamModule(_dbId);
+    await this.ensureModules(_dbId);
```
Comment (aguillenv) [inline src/modules/aws_ecr/rpcs/build.ts:59]: can we use `dbId` instead of `_dbId`? Let's follow the convention that when the parameter is prefixed with `_` is because it is not in use.

File: src/router/db.ts [hunk 7]
```diff
@@ -249,57 +249,56 @@ async function rpcCall(
   postTransaction?: PostTransactionCheck,
 ): Promise<any[] | undefined> {
   let rpcRes: any[] | undefined;
-  let errorOcurred = false;
-  let finallyErr;
+  let rpcErr;
```
Comment (aguillenv) [inline src/router/db.ts:252]: well, changes in this file should not be here since they have not been approved/merged. I need to fix a new bug that David spot deploying to staging so that will delay the changes.

File: src/modules/aws_ecr/rpcs/build.ts (hunk 5 above)
Comment (mtp1376) [reply src/modules/aws_ecr/rpcs/build.ts:126]: Can you clarify this a bit?

File: src/modules/aws_ecr/rpcs/build.ts (hunk 6 above)
Comment (mtp1376) [reply src/modules/aws_ecr/rpcs/build.ts:59]: Sure.

File: src/modules/aws_ecr/rpcs/build.ts (hunk 5 above)
Comment (mtp1376) [reply src/modules/aws_ecr/rpcs/build.ts:126]: If you mean parentheses, the linter will remove them once added.

File: examples/ecs-fargate/django/app/infra/migrations/0001_install.py (hunk 4 above)
Comment (mtp1376) [reply examples/ecs-fargate/django/app/infra/migrations/0001_install.py:17]: https://github.com/iasql/iasql-engine/pull/1682/files#diff-8818acbe0b7c48eb9a08c6d295a3de62882605533c5ebab4ed6b23bab3315f4aR149
It is used in `ecr_build` therefore we should install `aws_codebuild` to use it.

File: src/router/db.ts (hunk 7 above)
Comment (mtp1376) [reply src/router/db.ts:252]: I checked that file out from `main` branch.

File: src/modules/aws_codebuild/rpcs/import_source_credential.ts (hunk 3 above)
Comment (mtp1376) [reply src/modules/aws_codebuild/rpcs/import_source_credential.ts:39]: Added.

**Review by dfellis — APPROVED** [review]

File: examples/ecs-fargate/django/app/infra/migrations/0001_install.py (hunk 4 above)
Comment (aguillenv) [reply examples/ecs-fargate/django/app/infra/migrations/0001_install.py:17]: oh, I see. Any reason to go with this pattern and not a hard dependency on the module?

**Review by aguillenv — APPROVED** [review]

File: examples/ecs-fargate/django/app/infra/migrations/0001_install.py (hunk 4 above)
Comment (mtp1376) [reply examples/ecs-fargate/django/app/infra/migrations/0001_install.py:17]: The module works without those modules installed (read/update/create/delete), so I thought we don't need to add the overhead to sync/apply by adding another module. Just the RPC depends on it so added the soft-dependency.
